README.md

```diff
@@ -22,26 +41,183 @@ allprojects {
+                    is ErrorResult -> {
+                        // Handle error
+                    }
+                    is Success -> {
+                        // Handle success
+                    }
+                }
+            }
+```
+
+## Track a custom event
+
+Once you've identified a customer, you can use the `track` method to capture customer activity and send it Customer.io, optionally supplying event data with your request
+
+```kotlin
+CustomerIO.instance().track(
+                name = "logged_in",
+                attributes = mapOf("ip" to "127.0.0.1")
+            ).enqueue { result ->
+                when (result) {
+                    is ErrorResult -> {
+                        // Handle error
+                    }
+                    is Success -> {
+                        // Handle success
+                    }
+                }
+            }
+```
+
+`attributes` accepts the following data types:
+- Primitives (int, float, char...) and their boxed counterparts (Integer, Float, Character...).
+- Arrays, Collections, Lists, Sets, and Maps
+- Strings
+- Enums
 
-Thank you for your interest in wanting to contribute to the project! Let's get your development environment setup so you can get developing.
+## Stop identifying a customer
+In your app you may need to stop identifying a profile in the Customer.io SDK. There are 2 ways to do that:
+
+1. Call the `identify()` function which will stop identifying the previously identified profile (if there was one) and remember this new profile.
+2. Use `clearIdentify()` to stop identifying the previously identified profile (if there was one). 
+
+```kotlin
+CustomerIO.instance().clearIdentify()
+```
+
+# Push notifications 
+Want to send push notification messages to your customer's devices? Great!
+
+The Customer.io SDK supports sending push notifications via FCM for now. 
+
+## Getting Started FCM 
+
+### Dependency
+Push messaging artifact is needed in order to use push notification features
+```groovy
+implementation 'io.customer.android:messaging-push-fcm:<version-here>'
+```
+
+The artifact adds a `FirebaseMessagingService` to the app manifest automatically, so you don't have to do any extra setup to handle incoming push messages.
+
+In case, your application implements its own `FirebaseMessagingService`, make sure on  `onMessageReceived` and `onNewToken` method calls you additionally call CustomerIOFirebaseMessagingService.onMessageReceived and CustomerIOFirebaseMessagingService.onNewToken, respectively:
+
+```kotlin
+class FirebaseMessagingService : FirebaseMessagingService() {
+
+ override fun onMessageReceived(message: RemoteMessage) {
+        val handled = CustomerIOFirebaseMessagingService.onMessageReceived(this, message)
+        if (handled) {
+            logger.breadcrumb(this, "Push notification has been handled", null)
+        }
+ }
+ 
+ override fun onNewToken(token: String) {
+     CustomerIOFirebaseMessagingService.onNewToken(token)
```

Comment:
I suggest a note in the doc mentioning that at this time, the SDK [does not have error handling implemented](https://github.com/customerio/customerio-android/blob/c447a38b6d47d28b5200b23b855909f52eb0fb43/messagingpush/src/main/java/io/customer/messagingpush/CustomerIOFirebaseMessagingService.kt#L185) so if there is an error adding the device token to the CIO profile, the device token will be lost.

Reply:
The code sample in the README has been updated to use `onNewToken()` with an error callback so this comment has been addressed 👍